From fb6470ce380b8d6efc8a457b65a6dffd10691bfa Mon Sep 17 00:00:00 2001
From: Pavan Kumar Ramayanam <pavan.ramayanam@amd.com>
Date: Wed, 6 Dec 2017 19:21:43 +0530
Subject: [PATCH] drm/amd/pp: Read the maximum clock frequency from  pplib
 using pp_dpm_get_sclk() and pp_dpm_get_mclk() incase of failure when  using
 phm_get_max_high_clocks()

Change-Id: I40368191a98165b6f93ed35f3ff46a41f065b174
Signed-off-by: Raveendra Talabattula <raveendra.talabattula@amd.com>

Signed-off-by: Pavan Kumar Ramayanam <pavan.ramayanam@amd.com>
---
 drivers/gpu/drm/amd/powerplay/amd_powerplay.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/amd/powerplay/amd_powerplay.c b/drivers/gpu/drm/amd/powerplay/amd_powerplay.c
index f73e80c..ae9248e 100644
--- a/drivers/gpu/drm/amd/powerplay/amd_powerplay.c
+++ b/drivers/gpu/drm/amd/powerplay/amd_powerplay.c
@@ -1474,6 +1474,11 @@ int amd_powerplay_get_display_mode_validation_clocks(void *handle,
 		ret = phm_get_max_high_clocks(hwmgr, clocks);
 
 	mutex_unlock(&pp_handle->pp_lock);
-	return ret;
-}
+	
+	if (ret) {
+		clocks->memory_max_clock = pp_dpm_get_mclk(handle, false);
+		clocks->engine_max_clock = pp_dpm_get_sclk(handle, false);
+	}
 
+	return 0;
+}
-- 
2.7.4

